docs/user-guide/src/bmo/ironic_installation.md

```diff
@@ -0,0 +1,181 @@
+# Install Ironic
+
+Baremetal Operator must be able to interact with Ironic. First
+we will see what are the Ironic optional containers we need to
+run. After we will go through Ironic installation in-cluster and
+out-of-cluster. In any case, there are seven containers that need
+to run to be able to provision baremetal nodes.
+
+- ironic-dnsmasq
+- ironic-api
+- ironic-conductor
+- ironic-log-watch
+- ironic-inspector
+- ironic-ipa-downloader
+- mariadb
+
+To know more about each container's functionality check the documentation
+[here](https://github.com/metal3-io/ironic-image#description).
+
+## Prerequisites
+
+Container runtime (e.g., docker, podman). Below we use docker.
+
+<!-- include these inline so we can style an match variables -->
+{{#include ironic_variables.md}}
+
+## Optional containers
+
+In some circumstances, it might be necessary to install some additional containers on
+top of Ironic containers. And this might depend on whether your target nodes are
+virtual or physical machines, or whether you want to use more stable BMC protocol
+like Redfish and etc. Let's check why we would need these optional containers.
+We run these optional containers on the host machine, i.e., out-of-cluster regardless
+of the Ironic installation method.
+
+### vbmc
+
+If you plan to manage [livbirt](https://en.wikipedia.org/wiki/Libvirt) Virtual Machine (VM),
+there is a high chance that you need to install Virtual Baseboard Management
+Controller (VBMC), which is virtual implementation of Baseboard Management
+Controller (BMC). For More info on VBMC check official [documentation](https://docs.openstack.org/virtualbmc/latest/).
+
+```bash
+ sudo docker run -d --net host \
+    --name vbmc \
+     -v "SOME_LOCAL_PATH":/root/.vbmc \
+     -v "/root/.ssh":/root/ssh \
+     quay.io/metal3-io/vbmc:latest
+```
+
+Our VMBC container image is available at [quay.io/metal3-io/vbmc](https://quay.io/repository/metal3-io/vbmc?tag=latest&tab=tags)
+
+### sushy-tools
+
+Sushy-tools is the same functionality as VBMC tool, except the fact it relies on [Redfish](https://www.dmtf.org/standards/redfish)
+protocol instead of IPMI. For more info, check official [documentation](https://docs.openstack.org/sushy-tools/latest/index.html).
```

Comment:
```suggestion
protocol instead of IPMI. For more info, check the official [documentation](https://docs.openstack.org/sushy-tools/latest/index.html).
```